policy-test/tests/accepted_http_route_status.rs

```diff
@@ -0,0 +1,216 @@
+use kube::ResourceExt;
+use linkerd_policy_controller_k8s_api as k8s;
+use linkerd_policy_test::{await_route_status, create, with_temp_ns};
+
+#[tokio::test(flavor = "current_thread")]
+async fn inbound_accepted_parent() {
+    with_temp_ns(|client, ns| async move {
+        // Create a test 'Server'
+        let server = k8s::policy::Server {
+            metadata: k8s::ObjectMeta {
+                namespace: Some(ns.to_string()),
+                name: Some("test-server".to_string()),
+                ..Default::default()
+            },
+            spec: k8s::policy::ServerSpec {
+                pod_selector: k8s::labels::Selector::from_iter(Some(("app", "test-server"))),
+                port: k8s::policy::server::Port::Name("http".to_string()),
+                proxy_protocol: Some(k8s::policy::server::ProxyProtocol::Http1),
+            },
+        };
+        let server = create(&client, server).await;
+        let srv_ref = vec![k8s::policy::httproute::ParentReference {
+            group: Some("policy.linkerd.io".to_string()),
+            kind: Some("Server".to_string()),
+            namespace: server.namespace(),
+            name: server.name_unchecked(),
+            section_name: None,
+            port: None,
+        }];
+
+        // Create a route that references the Server resource.
+        let _route = create(
+            &client,
+            mk_inbound_route(&ns, "test-accepted-route", Some(srv_ref)),
+        )
+        .await;
+        // Wait until route is updated with a status
+        let route_status = {
+            let status = await_route_status(&client, &ns, "test-accepted-route").await;
+            status
+                .parents
+                .get(0)
+                .expect("must have at least one parent status")
+                .clone()
+        };
+
+        // Check status references to parent we have created
+        let parent_ref = &route_status.parent_ref;
+        assert_eq!(parent_ref.group, Some("policy.linkerd.io".to_string()));
+        assert_eq!(parent_ref.kind, Some("Server".to_string()));
+        assert_eq!(parent_ref.name, server.name_unchecked());
+
+        // Check status is accepted with a status of 'True'
+        let cond = route_status
+            .conditions
+            .iter()
+            .find(|cond| cond.type_ == "Accepted".to_string())
```

Comment:
see the clippy suggestion

Reply:
Weirdly, clippy suggests dereferencing this. I instead just removed the to_string conversion and it worked.